formatTest/unit_tests/input/syntax.re

```diff
@@ -864,6 +864,10 @@ let A | B | C = X;
  */
 external f : int => int = "foo";
 
+external foo : label::(<foo : int>)? => unit = "bla";
```

Comment:
I think the correct fix is to actually allow `<..>` as a "simple type" which should allow it to be parsed after the `::`. Did you try that?